AnkiDroid/src/main/java/com/ichi2/anki/provider/CardContentProvider.java

```
@@ -927,8 +927,35 @@ public Uri insert(Uri uri, ContentValues values) {
                 // Model ID is generated automatically by libanki
                 throw new IllegalArgumentException("Not possible to insert model with specific ID");
             case MODELS_ID_TEMPLATES:
-                // Adding new templates after the model is created could require a full-sync
-                throw new IllegalArgumentException("Templates can only be added at the time of model insertion");
+                Models models = col.getModels();
+                Long mid = getModelIdFromUri(uri, col);
+                JSONObject existingModel = models.get(mid);
+                if (existingModel == null) {
+                    throw new IllegalArgumentException("model missing: " + mid);
+                }
+                String name = values.getAsString(CardTemplate.NAME);
+                String qfmt = values.getAsString(CardTemplate.QUESTION_FORMAT);
+                String afmt = values.getAsString(CardTemplate.ANSWER_FORMAT);
+                String bqfmt = values.getAsString(CardTemplate.BROWSER_QUESTION_FORMAT);
+                String bafmt = values.getAsString(CardTemplate.BROWSER_ANSWER_FORMAT);
+                try {
+                    JSONObject t = models.newTemplate(name);
+                    try {
+                        t.put("qfmt", qfmt);
+                        t.put("afmt", afmt);
+                        t.put("bqfmt", bqfmt);
+                        t.put("bafmt", bafmt);
+                    } catch (JSONException e) {
+                        throw new RuntimeException(e);
+                    }
+                    models.addTemplate(existingModel, t);
+                    col.getModels().save(existingModel);
+                    return ContentUris.withAppendedId(uri, t.getInt("ord"));
```

I got around it using my old trick:

1. Run the connectedCheck and let it fail
2. adb install -r AnkiDroid/build/outputs/apk/AnkiDroid-debug.apk
3. manually set storage and ankidroid permission on AnkiDroid app
4. Rerun the connectedCheck